Code/GraphMol/test1.cpp

```diff
@@ -1574,6 +1574,84 @@ void testGithub1843() {
   BOOST_LOG(rdErrorLog) << "Finished" << std::endl;
 }
 
+void testHasValenceViolation() {
+  BOOST_LOG(rdInfoLog) << " ----------> Testing Atom::hasValenceViolation()"
+                       << std::endl;
+  auto to_mol = [](const auto &smiles) {
+    int debugParse = 0;
+    bool sanitize = false;
+    auto mol = RDKit::SmilesToMol(smiles, debugParse, sanitize);
+    TEST_ASSERT(mol != nullptr);
+    mol->updatePropertyCache(false);
+    return mol;
+  };
+
+  //  All valid atoms
+  for (const auto &smiles : {
+           "C",
+           "C(C)(C)(C)C",
+           "S(C)(C)(C)(C)(C)C",
+           "O(C)C",
+           "[H-]",
+           "[H+] |^2:0|",  // same as [H-]
```

Comment:
Just to check, does this make sense? Are these indeed equivalent?

Reply:
I think this is an instance of a non-physical radical count: H+ has no electrons, so it can't have a radical associated with it.

Reply:
Updated so this is a valence error